h5netcdf/core.py

```diff
@@ -234,6 +239,46 @@ def __init__(self, parent, name):
                     if k.startswith('_nc4_non_coord_'):
                         var_name = k[len('_nc4_non_coord_'):]
                     self._variables.add(var_name)
+
+        # One last pass to get the current sizes. This is necessary as h5netcdf
+        # needs to now the current size of each unlimited dimension.
+        #
+        # I'm not entirely sure how the netcdf C api determines the current
+        # size - should probably be looked into.
+        def _find_dim(h5group, dim):
+            if dim not in h5group:
+                # Might happen if the dimension has not been created yet.
+                if h5group.name == "/":
+                    return None
```

Comment:
I don't think this can happen here. We are only checking dimensions in `self.dimensions`, which looks like at `self._dim_sizes` -- which in turn is only filled in where there is already a matching dimension scale.

It would be better to raise an error here (maybe `AssertionError`?) because this really shouldn't ever happen.